oak_runtime/src/node/http/mod.rs

```diff
@@ -488,24 +497,26 @@ impl HttpResponseIterator {
         response_receiver.receive(&self.runtime)
     }
 
-    fn to_response(&self) -> Response<Body> {
+    fn to_response(&self) -> Result<Response<Body>, http::Error> {
         info!(
             "Generating response for runtime {} and reader {:?}.",
             self.runtime.node_id.0, self.response_reader
         );
-        let mut response = Response::new(Body::empty());
         match self.read_response() {
             Ok(http_response) => {
                 let status_code = http_response.status as u16;
-                *response.body_mut() = Body::from(http_response.body);
-                *response.status_mut() = StatusCode::from_u16(status_code)
+                let status = StatusCode::from_u16(status_code)
```

Comment:
(nit) the naming seems reversed here, it seems that `status_code` and `status` should be swapped?

Reply:
I used status because the method name is `status`. I guess both could be `status_code`.